mpf/devices/shot.py

```diff
@@ -412,6 +414,23 @@ def _delay_switch_hit(self, switch_name, **kwargs):
     def _release_delay(self, switch):
         self.active_delays.remove(switch)
 
+    def _register_jump_state_handlers(self):
+        for jump_state in self.config['jump_states']:
+            for event in jump_state['events']:
+                self._handlers.append(self.machine.events.add_handler(event, self._jump_states,
+                                                                      jump_state_config=jump_state))
+
+    def _remove_jump_state_handlers(self):
+        self.machine.events.remove_handlers_by_keys(self._handlers)
+
+    @event_handler(7)
+    def _jump_states(self, jump_state_config, **kwargs):
+        """Takes in a jump_state event to move the state of the given shot to a specific state. (Default = State 0)"""
```

Comment:
You note the default state is 0, yet there are no defaults in this handler. I see that the config_spec sets 0 as the default value but I feel like the method documentation shouldn't make assumptions about other areas of the code. Would you update the handler, or update the comment?

Reply:
Agreed and good catch.  I cleaned up to note state a default, since there truly isn't.  I guess there should be error handling on the processing of it?  If the state doesn't have a value, then throw an error that its missing a required field.  If this seems like a good idea, can you point me somewhere to use as a reference in how to message that, or require during config validation.

Reply:
That was horribly worded.  To make sense of it.  It gives the general config error if I remove 0 as a default value.  Do we think it would be better to give you an error to make it clear that you should be telling it what state?  Or do we think it would be better to plug a value, even if its wrong?

My vote would be to remove the config_spec of 0 so that a user gets an error that they are missing a required argument per the standard cmd warning.  Obviously once this is approved, I'll update the docs to include that events and state are the two required fields.

Reply:
I have updated to remove the default value.  This will be flagged as a required field.  Once approved, I will write the Help Documents that list event and state as required fields with no defaults.